public/app/features/browse-dashboards/state/hooks.ts

```diff
@@ -126,19 +126,27 @@ export function useLoadNextChildrenPage(folderUID: string | undefined) {
  * @param openFolders Object of UID to whether that item is expanded or not
  * @param level level of item in the tree. Only to be specified when called recursively.
  */
-function createFlatTree(
+export function createFlatTree(
   folderUID: string | undefined,
   rootCollection: BrowseDashboardsState['rootItems'],
   childrenByUID: BrowseDashboardsState['childrenByParentUID'],
   openFolders: Record<string, boolean>,
-  level = 0
+  level = 0,
+  insertEmptyFolderIndicator = true
```

Comment:
is this needed anymore?

Reply:
Folder picker uses it to not get the "empty folder" rows https://github.com/grafana/grafana/blob/joshhunt/nested-folder-picker/public/app/core/components/NestedFolderPicker/NestedFolderPicker.tsx#L62

Reply:
ahhhhh got it, that was the logic i was missing ty 👍